core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -726,14 +729,27 @@ public SessionBuilder setSystemProperties(Map<String, String> systemProperties)
             return this;
         }
 
+        /**
+         * Extracts processed connector properties from provided session and sets them in the builder.
+         * An explicit set method is not added for this field to prevent SessionBuilder callers
+         * from doing so without proper validation. Any previously set connectorProperties are discarded.
+         */
+        public SessionBuilder setProcessedConnectorPropertiesFrom(Session session)
+        {
+            requireNonNull(session, "session is null");
+            this.connectorProperties.clear();
+            this.connectorProperties.putAll(session.getConnectorProperties());
+            return this;
+        }
```

Comment:
Just call this method `setConnectorProperties`.

I'd also add a comment like "no need to copy the property maps because they are already immutable in the Session" to avoid anyone trying to "fix" it later.

Reply:
> Just call this method setConnectorProperties.

`setCatalogProperty` usage is common, and accessors don't need to worry about validation. To avoid confusion with that method, I added `processed` in the name (and kept the other method less verbose). Also, suffixing the name with `From` seems more readable to me as without that, one would expect the parameter and field types to be the same.

Reply:
added comment.

Reply:
I don't think there's much of a risk of confusion between `setConnectorProperties` and `setCatalogProperty`,<sup>1</sup> except which properties go where, which would probably be better clarified in documentation.

I prefer it without `From` because an IDE will help you with the parameter type, but I don't mind it either way.

(I also like `copyConnectorProperties`, but that might be less obvious to people expecting `set`.)

<sup>1</sup> Well, `setCatalogSessionProperty`, which should probably be renamed to `setCatalogProperty` in a separate PR.

Reply:
@jirassimok I think the difference in the wording for `setConnectorProperties` and `setCatalogProperty` doesn't indicate difference in semantics properly. Feel free to suggest if there's a better naming suggestion to communicate the semantics other than including "processed" in the new setter. cc @findepi if you've any preference here, or any ideas/concerns around the method signature itself.

Reply:
I agree that the semantics aren't indicated well by the names, but I don't think `unprocessed` correctly indicates the semantics, either, because based on the names alone, there isn't a clear relationship between connector and catalog properties in the first place.

Maybe the best way to go about making it clearer would be renaming the two fields and their corresponding methods to `connectorCatalogProperties` and `sessionCatalogProperties`?

Reply:
cc @electrum @dain for processed vs unprocessed